test/functional/mongo_client_tests.js

```diff
@@ -792,4 +797,27 @@ describe('MongoClient', function() {
       });
     }
   });
+
+  it('Should set optionsValidationLevel when validateOptions is set', {
+    metadata: { requires: { topology: ['single'] } },
+    // The actual test we wish to run
+    test: function() {
+      const configuration = this.configuration;
+      try {
+        configuration.newClient({}, { validateOptions: true, unknownOption: true });
+      } catch (err) {
```

Comment:
to me it looks like if `validateOptions` is true, the level should be [warn](https://github.com/mongodb/node-mongodb-native/pull/1895/files#diff-12bc6199a1348e0428b4b9ebeb881206L654) not `error`

Reply:
If `validateOptions` is true, [the level should be error.](https://github.com/mongodb/node-mongodb-native/pull/1895/files#diff-12bc6199a1348e0428b4b9ebeb881206L651) The warning is emitted by default.